### DISCUSSION TITLE
Sampling mesh points within a certain geodesic distance?

### DISCUSSION THREAD

**Discussion by lukkio88** [libigl/libigl#2178] [Q&A]
Hi,

I know libigl provides some functionality for geodesic distance (given mainly source and destination). I wonder however if there's a functionality (or if it is easy to implement) the following functionality:

Suppose you're given a set of point `p[0],...,p[N-1]` we want to sample `q[0],..,q[M-1]` randomly but such that for every `j` we have `min{ geodesic_distance(q[j],p[i]) < epsilon}` where `epsilon` is a user defined parameter.

Is there such functionality in libigl and if not can you maybe give a direction on how to use the library features to implement such a functionality? (hopefully in a easy way, not too easy I know).

A first thought I had was to take my source point and then as destination all the vertices of the mesh and calculate the geodesic distance (maybe using fast geodesic approximation, I don't think I need anything super accurate).

And then use such information together with uniform sampling of the mesh to accept only points within my distance threshold.
I am guessing however with this approach if my threshold is small I might get a lot of rejection so the sampling process might be too slow.

Thank you

**Comment by alecjacobson** [comment]
Do you mean `>epsilon`?

You might have a look at the `blue_noise` functions. These _don't_ use geodesic distance, but if the goal is a nice dense surface point distribution then they tend to work quite well for many models.

Using geodesic distance would be cool but I can see there might be challenges making it performant.

**Reply by lukkio88** [reply]
No I mean `< epsilon` cause I want the points within the given distance

**Reply by alecjacobson** [reply]
Hmm. I'm not sure this is enough to specify a distribution. I guess you could do binary search on the `n` input to `random_points_on_mesh` until the for each point the closest other point is <epsilon.

If epsilon is small relative to the mesh and there's not too much curvature/self-intersections then I would just use Euclidean distance since that's always a good approximation of geodesic if you're zoomed in enough.
